### PR TITLE
added retry logic to fix locking issue

### PR DESCRIPTION
**Pull request by SusmithaGundu**
fix for issue : https://devopsmx.atlassian.net/browse/OP-20665 and https://devopsmx.atlassian.net/browse/OP-20645

When running multpile requests to clouddriver /artifacts/fetch ,the threads trying to acquire the lock are being interrupted and the interrrupted exception is thrown. Added retry logic to try for 5 times to acquire lock before throwing exception

### CODE REVIEW

File: clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java [hunk 1]
```diff
@@ -64,17 +68,29 @@ public InputStream download(Artifact artifact) throws IOException {
...
-              + ").",
-          e);
+    int numberOfRetires = gitRepoArtifactProviderProperties.getRetry();
+    int retries = 0;
+    while (true) {
+      try {
+        return getLockedInputStream(
+            repoUrl, subPath, branch, stagingPath, repoBasename, outputFile);
+      } catch (InterruptedException e) {
+        if (retries == numberOfRetires) {
```
Comment (jsoundar22) [inline clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java:78]: Better to mention as >=

File: clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactProviderProperties.java [hunk 2]
```diff
@@ -32,5 +32,6 @@ public class GitRepoArtifactProviderProperties implements ArtifactProvider<GitRe
   private int cloneRetentionCheckMs = DEFAULT_CLONE_RETENTION_CHECK_MS;
   private long cloneRetentionMaxBytes = 1024 * 1024 * 100; // 100 MB
   private int cloneWaitLockTimeoutSec = 60;
+  private int retry = 5;
```
Comment (kirangodishala) [inline clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactProviderProperties.java:35]: can you rename to a more meaningful name like timedLockRetryCount, gitRepoFileLockRetryCount etc?

File: clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java [hunk 3]
```diff
@@ -64,17 +68,29 @@ public InputStream download(Artifact artifact) throws IOException {
...
+                  + branch
+                  + ").",
+              e);
+        }
+        try {
+          Thread.sleep(1000);
+        } catch (InterruptedException ex) {
+          log.debug("Interrupted by sleep method");
+        }
+        retries++;
```
Comment (kirangodishala) [inline clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java:92]: You could add an info/debug statement to indicate the retry count

File: clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java (hunk 1 above)
Comment (SusmithaGundu) [reply clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java:78]: When it hits the numberOfRetries itself the exception is thrown and execution stops

File: clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java (hunk 3 above)
Comment (SusmithaGundu) [reply clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java:93]: added retry count indication

File: clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java [hunk 4]
```diff
@@ -91,6 +91,8 @@ public InputStream download(Artifact artifact) throws IOException {
         }
         retries++;
         log.debug("retrying for {} time", retries);
+      } catch (Exception e) {
+        throw new IOException("encountered an exception while running ", e);
       }
```
Comment (kirangodishala) [inline clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java:96]: Additional catch is not needed....this won't catch the exceptions occurred within first catch block.

Comment (SusmithaGundu) [reply clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java:94]: removed it

**Pull request closed without merging**